### PR TITLE
New module Metrics with default algebra and basic usage

### LINKED ISSUE #11: Implement an algebra which calculate default metrics related with microservices
Implement an algebra which calculate the used memory, free memory, total memory and max memory of the own microservice. 

This algebra will be in a new module called Metrics and in the future will be migrate to a new project

### PR DESCRIPTION
**Pull request by eperinan**
It resolves #11 

I created a new module in the app which will be move to a new repository in the future. A part from that I created a module which will contain all the algebras related with metrics. At this moment I only created the Default algebra with the default metrics.

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -38,3 +38,19 @@ lazy val core = project
       %("joda-time")
     )
   )
+  .dependsOn(metrics)
+  .aggregate(metrics)
+
+lazy val metrics = project
+  .in(file("metrics"))
+  .settings(moduleName := "frees-metrics")
+  .settings(scalaVersion := "2.12.3")
```
Comment (juanpedromoreno) [inline build.sbt]: You shouldn't need this setting

File: core/src/main/scala/runtime/endpoints.scala [hunk 2]
```diff
@@ -19,52 +19,92 @@ package opscenter
...
+
+      private def signalFromClient: Sink[IO, WebSocketFrame] = _.evalMap { (ws: WebSocketFrame) =>
+        ws match {
+          case Text(t, _) => IO.apply(println(s"Received from client: $t"))
+          case f          => IO.apply(println(s"Received from client unknown type: $f"))
+        }
+      }
+
+      private def metricsStream: Stream[IO, WebSocketFrame] =
+        Scheduler[IO](2)
```
Comment (juanpedromoreno) [inline core/src/main/scala/runtime/endpoints.scala]: What's the number 2 mean?

File: metrics/src/main/scala/metrics.scala [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+package freestyle
+package metrics
+
+import freestyle._
+import freestyle.config.ConfigM
+import cats.effect.IO
+import cats.implicits._
+
+// Modules
```
Comment (juanpedromoreno) [inline metrics/src/main/scala/metrics.scala]: IMHO these comments don't bring value ;)

File: metrics/src/main/scala/metrics.scala [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+@module trait Metrics {
+  val default: Default
+}
+
+// Algebras
```
Comment (juanpedromoreno) [inline metrics/src/main/scala/metrics.scala]: Same here

File: metrics/src/main/scala/metrics.scala [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+@free trait Default {
```
Comment (juanpedromoreno) [inline metrics/src/main/scala/metrics.scala]: Maybe `DefaultMetrics`? or `DefaultMemoryMetrics`? In the latter case, we could get rid of the suffix `Memory` on each method. What do you think?

File: metrics/src/main/scala/runtime/default.scala [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+
+object implicits {
+
+  implicit def defaultHandler[M[_]](implicit C: Capture[M]): Default.Handler[M] =
+    new Default.Handler[M] {
+
+      private val mb      = 1024 * 1024
+      private val runtime = Runtime.getRuntime
+
+      def usedMemory: M[Float] = C.capture((runtime.totalMemory - runtime.freeMemory) / mb)
```
Comment (juanpedromoreno) [inline metrics/src/main/scala/runtime/default.scala]: Have you decided how units will be exposed? Do you think is better in MB?

File: metrics/src/main/scala/metrics.scala (hunk 3 above)
Comment (eperinan) [reply metrics/src/main/scala/metrics.scala]: Totally agree. Removed in the next commit. :)

File: metrics/src/main/scala/metrics.scala (hunk 4 above)
Comment (eperinan) [reply metrics/src/main/scala/metrics.scala]: Totally agree. Removed in the next commit. :)

File: metrics/src/main/scala/metrics.scala (hunk 5 above)
Comment (eperinan) [reply metrics/src/main/scala/metrics.scala]: Finally I decided `DefaultMemoryMetrics`. Thanks for the observation. Also I removed the suffix `Memory` from all methods.

File: build.sbt (hunk 1 above)
Comment (eperinan) [reply build.sbt]: I just removed the `scalaVersion`. Should I removed another settings more ?

File: core/src/main/scala/runtime/endpoints.scala (hunk 2 above)
Comment (eperinan) [reply core/src/main/scala/runtime/endpoints.scala]: That value is the `corePoolSize`. That value is the number of threads to keep in the pool, even if they are idle.

You can see the `apply` here. https://github.com/functional-streams-for-scala/fs2/blob/series/0.10/core/jvm/src/main/scala/fs2/SchedulerPlatform.scala#L13

File: metrics/src/main/scala/runtime/default.scala (hunk 6 above)
Comment (eperinan) [reply metrics/src/main/scala/runtime/default.scala]: Sorry, I think that it is better be consistent with the library `Runtime.java` and I am going to expose the metrics in bytes.

By default this library return all the measures in bytes.

Changed in the next commit

File: core/src/main/scala/runtime/endpoints.scala (hunk 2 above)
Comment (juanpedromoreno) [reply core/src/main/scala/runtime/endpoints.scala]: Maybe you could use named parameters clarify it.

**Review by juanpedromoreno — APPROVED** [review]
LGTM

File: core/src/main/scala/runtime/endpoints.scala (hunk 2 above)
Comment (eperinan) [reply core/src/main/scala/runtime/endpoints.scala]: Make sense. Changed. Thanks for the review buddy.

**Pull request merged**